### PR TITLE
Victor/bm rtos abstraction

### PR DESCRIPTION
**Pull request by victorsowa12**
WIP

### CODE REVIEW

File: src/lib/bcmp/CMakeLists.txt [hunk 1]
```diff
@@ -25,11 +25,15 @@ set(BCMP_FILES
     ${BCMP_DIR}/bm/bm_printf.cpp
     ${BCMP_DIR}/bm/bristlemouth.cpp
 
+    # BM FREERTOS - TODO turn this into a CMAKE variable?
+    ${BCMP_DIR}/bm_rtos/platform/bm_freertos.c
+
```
Comment (matt001k) [inline src/lib/bcmp/CMakeLists.txt:30]: I like this idea, we could have `-DRTOS_SELECT={your_fav_rtos_here}` as a cmake option to pass into the top level CMake directory of `bm_core` which would determine out of our supported RTOS what one to pick.

**Review by matt001k — COMMENTED** [review]
Nice work! I like this scheme that you have selected. Looks like unit tests need to be fixed, but other than that it looks good!

File: src/lib/bcmp/bm_rtos/bm_rtos.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+// Queue functions
+BmQueue bm_queue_create(uint32_t queue_length, uint32_t item_size);
+BmError bm_queue_receive(BmQueue queue, void *item, uint32_t timeout_ms);
+BmError bm_queue_send(BmQueue queue, const void *item, uint32_t timeout_ms);
+
+// Semaphore functions
+BmSemaphore bm_semaphore_create(void);
+BmError bm_semaphore_give(BmSemaphore semaphore);
+BmError bm_semaphore_take(BmSemaphore semaphore, uint32_t timeout_ms);
```
Comment (matt001k) [inline src/lib/bcmp/bm_rtos/bm_rtos.h:41]: Where is this defined at?

Comment (victorsowa12) [reply src/lib/bcmp/bm_rtos/bm_rtos.h:41]: Are you talking about what we talked about wrg to the:
```
#define MUTEX_UNLOCK (timeout) mutext_lock(timeout) {

#define MUTEX_LOCK mutex_lock()}
```
?

Comment (matt001k) [reply src/lib/bcmp/bm_rtos/bm_rtos.h:41]: No I do not see it's definition in the source file.

Comment (victorsowa12) [reply src/lib/bcmp/bm_rtos/bm_rtos.h:41]: woops! good eye!

**Comment by victorsowa12** [comment]
Closed as this work has been moved to https://github.com/bristlemouth/bm_common/pull/1

**Pull request closed without merging**